app/controllers/brapi/v1/phenotypes_controller.rb

```diff
@@ -0,0 +1,130 @@
+class Brapi::V1::PhenotypesController < Brapi::BaseController
+
+
+  attr_accessor :request_params, :user
+
+  rescue_from ActionController::ParameterMissing do |exception|
+    render json: { errors: { attribute: exception.param, message: exception.message } }, status: 422
+  end
+
+
+  def search
+    # accepted params: germplasmDbIds, observationVariableDbIds, studyDbIds, locationDbIds, seasonDbIds
+    # sortBy, sortOrder
+    # pageSize, page
+    # TODO Not supported yet: programDbIds, observationLevel
+
+    phenotypes_queries = Brapi::V1::PhenotypesQueries.instance
+    page = get_page
+    page_size = get_page_size 
+        
+    result_object = phenotypes_queries.phenotypes_search_query(
+      params['germplasmDbIds'], params['observationVariableDbIds'], params['studyDbIds'], 
+      params['locationDbIds'], params['seasonDbIds'], 
+      params['sortBy'], params['sortOrder'], page, page_size, false)
+    
+    records = result_object.values
+   
+    if records.nil? || records.size ==0
+      render json: { reason: 'Resource not found' }, status: :not_found  # 404
+    else
+      json_result_array = []
+      intermediate_hash = {}
+      
+      # any programmatic data manipulation can be done here
+      result_object.each do |row|
+        # To check authentication and ownership when ORCID is supported by BrAPI
+        # We currently only retrieve public records. This is already done at query level
+        #if (!row[:user_id] || row[:published] )
+        
+        intermediate_hash_id = row["observationUnitDbId"]
+        existing_row = intermediate_hash[intermediate_hash_id]
+        
+        # We introduce data related to plot numbers, plants, etc
+        begin
+          observation_levels_hash = JSON.load(row.delete("observation_levels_json"))
+          if observation_levels_hash != nil
+              row["plotNumber"]= observation_levels_hash["plot"]
+              row["plantNumber"]= observation_levels_hash["plant"]
+              row["blockNumber"]= observation_levels_hash["block"]
+              row["observationLevel"]= observation_levels_hash.keys[observation_levels_hash.keys.length-1]
+          end
+        rescue JSON::ParserError => e  
```

Comment:
observation_levels_json is based on design_factors.design_factors, which is free text introduced by the user, so it's not totally reliable.
In this case the exception only indicates the data was introduced in a different format than I need to manage it, but I think that format is not (still) mandatory when user introduces it, so I think I must not stop the normal execution of the procedure .